server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -150,56 +150,56 @@ private void extensionsDiscovery() throws IOException {
         if (!FileSystemUtils.isAccessibleDirectory(extensionsPath, logger)) {
             return;
         }
-        List<Path> pluginDirs = PluginsService.findPluginDirs(extensionsPath);
 
-        Set<Extension> extensions = new HashSet<Extension>();
-        if (!pluginDirs.isEmpty()) {
+        List<Extension> extensions = new ArrayList<Extension>();
+        if (Files.exists(extensionsPath.resolve("extensions.yml"))) {
             try {
                 extensions = readFromExtensionsYml(extensionsPath.resolve("extensions.yml")).getExtensions();
             } catch (Exception e) {
-                throw new IOException("Could not read from extensions.yml");
+                throw new IOException("Could not read from extensions.yml\n " + e);
```

Comment:
Why not use [`IOException(message, throwable)`](https://docs.oracle.com/javase/7/docs/api/java/io/IOException.html#IOException(java.lang.String,%20java.lang.Throwable)) syntax here?  (Not sure if that's better, just asking.)

Also per earlier comment, if `readFromExtensionsYml` is throwing `IOException` what is the value add of catching it just to re-throw it?